src/mlpack/methods/ann/layer/transposed_convolution_impl.hpp

```diff
@@ -57,18 +57,23 @@ TransposedConvolution<
     const size_t kernelHeight,
     const size_t strideWidth,
     const size_t strideHeight,
-    const size_t padWidth,
-    const size_t padHeight,
+    const size_t padW,
+    const size_t padH,
     const size_t inputWidth,
     const size_t inputHeight,
     const size_t outputWidth,
-    const size_t outputHeight) :
+    const size_t outputHeight,
+    const std::string& paddingType) :
```

Comment:
do you think it is possible to call the other overloaded constructor to avoid the duplicate code?

Reply:
Great idea, I will switch to calling the the other constructor.

Reply:
Small issue, the constructor checks for correct output dimensions and the first constructor has padWLeft == padWRight, however there might be a case such that unequal padding results in desired output shape but unequal might not and the first constructor will throw an  error, So I am keeping it as is.

Reply:
If I understood you correctly, I wanted the first constructor to call the second constructor and not the other way around.

Reply:
Ohh, I was trying the other way round, I'll make the changes.

Reply:
Hi @saksham189, I have removed the duplicated portion of code across all layers, Have a look when you get the chance. Also, Thanks for the idea.